external-crates/move/crates/move-compiler/src/typing/core.rs

```diff
@@ -1396,8 +1672,17 @@ pub fn instantiate(context: &mut Context, sp!(loc, t_): Type) -> Type {
         Apply(abilities_opt, n, ty_args) => {
             instantiate_apply(context, loc, abilities_opt, n, ty_args)
         }
+        Fun(args, result) => Fun(
+            args.into_iter().map(|t| instantiate(context, t)).collect(),
+            Box::new(instantiate(context, *result)),
+        ),
         x @ Param(_) => x,
-        Var(_) => panic!("ICE instantiate type variable"),
+        // instantiating a var really shouldn't happen... but it does because of macro expansion
+        // We expand macros before type checking, but after the arguments to the macro are type
+        // checked (otherwise we couldn't properly do method syntax macros). As a result, we are
+        // substituting type variables into the macro body, and might hit one while expanding a
+        // type in the macro where a type parameter's argument had a type variable.
+        x @ Var(_) => x,
```

Comment:
While the current macro expander doesn't end up in a situation where it might occur (since recoloring is separated from substitution), it may be nice to also allow vars in `subst_tparams` by a similar argument.